FirebaseAuth/Sources/Swift/AuthProvider/PhoneAuthProvider.swift

```diff
@@ -86,9 +86,9 @@ import Foundation
               uiDelegate: uiDelegate,
               multiFactorSession: multiFactorSession
             )
-            Auth.wrapMainAsync(callback: completion, withParam: verificationID, error: nil)
+            Auth.wrapMainAsync(callback: completion, with: .success(verificationID ?? ""))
```

Comment:
This PR surfaced one vulnerability in this method where the async version could possibly crash with if `nil, nil` is encountered. 

I don't think falling back to an empty string like I do here makes sense though so I'm investigating how to make it non-optional.

Reply:
Moved to throwing errors.